docs/contributor/arch/008-telemetry-flow-healthiness-status-api.md

```diff
@@ -0,0 +1,102 @@
+
+### Exporter Queueing
+
+* All batches are enqueued first.
+* Queue fills up when consumers are slower than producers. It can be caused by backend issues or a mismatch between the ingestion and export rate (e.g. backend is slow).
+* If the exporter queue is full, data is dropped (`otelcol_exporter_enqueue_failed_metric_points` goes up).
+* Watch for high queue size: `otelcol_exporter_queue_size / otelcol_exporter_queue_capacity > THRESHOLD`.
+
+### Exporter Retries
+
+* For non-retryable errors, data is dropped. See more about retryable and non-retryable errors.
+* Retryable errors trigger retries until success or retry limit, then data is dropped.
+* `otelcol_exporter_send_failed_metric_points` is increased in both scenarios if data is dropped.
+* If data is sent successfully, `otelcol_exporter_sent_metric_points` is increased.
+
+## Decision
+
+We are choosing between two alternatives:
+
+* Using multiple condition types to represent various telemetry flow events (Throttling, Data Loss, High Buffer Utilization, etc.).
+* Using a single condition type (TelemetryFlowHealthy) with a reason field to denote diverse telemetry flow events.
+
+After careful consideration, we have opted for the single condition type TelemetryFlowHealthy as it minimizes cognitive load for the end user.
+Ultimately, the user's primary concern is understanding whether the telemetry flow is functioning correctly. In case of any issues, the user has the following actionable steps:
+
+* Troubleshoot the backend.
+* Investigate backend connectivity.
+* Reduce ingestion.
+* Manually scale out the gateway (as long as no autoscaling capability is provided).
+
+In this scenario, we can assign the reason field a value that holds utmost significance for the user:
+```
+FullDataLoss > PartialDataLoss > HighBufferUtilization > GatewayThrottling > Healthy
+```
+
+The reasons will then be based on the following alert rules (an example for Metric Pipelines the actual PromQL expressions have to be defined later):
+| Alert Rule | Expression |
+| --- | --- |
+| GatewayExporterSentMetrics    | `sum(rate(otelcol_exporter_sent_metric_points{...}[5m])) > 0`           |
+| GatewayExporterDroppedMetrics  | `sum(rate(otelcol_exporter_send_failed_metric_points{...}[5m])) > 0`    |
+| GatewayReceiverRefusedMetrics  | `sum(rate(otelcol_receiver_refused_metric_points{...}[5m])) > 0`        |
+| GatewayExporterEnqueueFailed   | `sum(rate(otelcol_exporter_enqueue_failed_metric_points{...}[5m])) > 0` |
+| GatewayExporterQueueAlmostFull | `otelcol_exporter_queue_size / otelcol_exporter_queue_capacity > 0.8`   |                               |
+
+We could then map the alert rules to the reasons as follows:
+
+| Reason | Alert Rules |
+| --- | --- |
+| FullDataLoss           | **not** GatewayExporterSentMetrics **and** (GatewayExporterDroppedMetrics **or** GatewayExporterEnqueueFailed) |
```

Comment:
```suggestion
| CompleteDataLoss           | **not** GatewayExporterSentMetrics **and** (GatewayExporterDroppedMetrics **or** GatewayExporterEnqueueFailed) |
```
`FullDataLoss` sounds less professional

Reply:
If you like my suggestions in line 64, let's repeat them here.